Problems with case matching and user organizations

Hi @ViViDboarder, thanks for the taking the time to develop this tool. I managed to get it up and running, but I'm having some odd errors I'm hoping you can help me with. I have the LDAP filters such that only a single user should match for testing purposes, and that is working.

The first time it ran, I got the following log output (redacted for identities):

`Existing user or invite found with email: user.one@domain.com,`
`Existing user or invite found with email: user.two@domain.com,`
`Existing user or invite found with email: user.three@domain.com,`
`Try to invite user: User.Four2@domain.com,`
`Sent invites to 1 user(s).,`

Which is working great so far. The user received their invitation, and I see the user in bitwarden_rs's admin panel. However, the user is not associated with any current organizations in our setup, so my first question is: is there a way to specify which organizations new users should be associated with?

Then, on the next cycle I saw:

`Existing user or invite found with email: user.one@domain.com,`
`Existing user or invite found with email: user.two@domain.com,`
`Existing user or invite found with email: user.three@domain.com,`
`Existing user or invite found with email: user.four2@domain.com,`
`Try to invite user: User.Four2@domain.com,`
`Sent invites to 1 user(s).,`

I note that it's trying to reinvite the same user. I wonder if it isn't because our AD server returns the email address with capital letters in it, and that the matching is being done on a case sensitive level? I note the following in bitwarden_rs's logs seem to reflect the error in attempting to duplicate an existing user:

`[2020-03-05 15:15:28][request][INFO] GET /admin/users`
`[2020-03-05 15:15:28][response][INFO] GET /admin/users (get_users) => 200 OK,`
`[2020-03-05 15:15:28][request][INFO] POST /admin/invite,`
`[2020-03-05 15:15:28][error][ERROR] User already exists,`
`[2020-03-05 15:15:28][response][INFO] POST /admin/invite (invite_user) => 400 Bad Request,`

So my second question is: is there a way to fix the case sensitivity of the matching?

Thanks!

--- a/src/main.rs
+++ b/src/main.rs
@@ -42,7 +42,7 @@ fn get_existing_users(client: &mut bw_admin::Client) -> Result<HashSet<String>, 
     let all_users = client.users()?;
     let mut user_emails = HashSet::with_capacity(all_users.len());
     for user in all_users {
-        user_emails.insert(user.get_email());
+        user_emails.insert(user.get_email().to_lowercase());
         if user.is_disabled() {
             println!(
                 "Existing disabled user found with email: {}",
@@ -124,7 +124,7 @@ fn invite_from_ldap(
                     .get(mail_field.as_str())
                     .and_then(|l| (l.first()))
                 {
-                    if existing_users.contains(user_email) {
+                    if existing_users.contains(&user_email.to_lowercase()) {
                         println!("User with email already exists: {}", user_email);
                     } else {
                         println!("Try to invite user: {}", user_email);

